src/systems/ec/implementations/react/packages/link/Link.jsx

```diff
@@ -1,44 +1,69 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import React from 'react';
 import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Icon from '@ecl/ec-react-component-icon/Icon';
 
 const Link = ({
   variant,
   href,
   label,
-  // icon,
-  // iconPosition,
-  extraClasses,
-  extraAttributes,
+  icon,
+  iconPosition,
+  className,
   ...props
 }) => {
-  let classes = 'ecl-link';
-  if (variant) classes += ` ecl-link--${variant}`;
-  if (extraClasses) classes += ` ${extraClasses}`;
+  const classNames = classnames(className, {
+    'ecl-link': true,
+    [`ecl-link--${variant}`]: variant,
+    [`ecl-link--icon-${iconPosition}`]: icon && icon.icon,
+  });
+
+  let link = '';
+  if (icon && icon.icon) {
+    if (iconPosition === 'before') {
+      link = (
+        <a {...props} href={href} className={classNames}>
+          <Icon {...icon} className="ecl-link__icon" />
```

Comment:
`icon.className` is lost on the way. 

-> `<Icon {...icon} className={classnames(icon.className, 'ecl-link__icon')} />`